docs/kubeadm.md

```diff
@@ -0,0 +1,249 @@
+    deb http://packages.cloud.google.com/apt kubernetes-xenial-unstable main
+    EOF
+    # apt-get update
+    # apt-get install -y kubelet-kubeadm kubectl
+    # # OR
+    # apt-get install -y docker-engine=1.11.2-0~xenial socat
+-->
+
+
+* SSH into the machine and become `root` if you are not already (for example, run `sudo su -`).
+* If the machine is running Ubuntu 16.04, run:
+
+      # curl -sSL https://get.docker.com/ | sh
+      # apt-get install -y socat
+      # curl -s -L \
+        "https://www.dropbox.com/s/shhs46bzhex7dxo/debs-9b4337.txz?dl=1" | tar xJv
+      # dpkg -i debian/bin/*.deb
+      # # XXX https://github.com/kubernetes/release/issues/98
+      # systemctl daemon-reload && systemctl restart kubelet
+
+   If the machine is running Fedora 24, run:
+
+      # cat <<EOF > /etc/yum.repos.d/k8s.repo
+      [kubelet]
+      name=kubelet
+      baseurl=http://files.rm-rf.ca/rpms/kubelet/
+      enabled=1
+      gpgcheck=0
+      EOF
+      # setenforce 0
+      # yum install kubelet kubeadm kubectl kubernetes-cni
+      # systemctl enable docker && systemctl start docker
+      # systemctl enable kubelet && systemctl start kubelet
+
+The kubelet will now be restarting every few seconds, as it waits in a crashloop for `kubeadm` to tell it what to do.
+
+Optionally, see also [more details on installing Docker](https://docs.docker.com/engine/installation/#/on-linux).
+
+### (2/4) Initializing your master
+
+The master is the machine where the "control plane" components run, including `etcd` (the cluster database) and the API server (which the `kubectl` CLI communicates with).
+All of these components will run in containers started by `kubelet`.
+
+To initialize the master, pick one of the machines you previously installed `kubelet` and `kubeadm` on, and run:
+
+* If you want to be able to schedule workloads on the master, for example if you want a single-machine Kubernetes cluster for development, run:
+
+      # kubeadm init --schedule-workload
+
+* If you do not want to be able to schedule workloads on the master (perhaps for security reasons), run:
+
+      # kubeadm init
+
+This will download and install the cluster database and "control plane" components.
+This may take several minutes.
+
+The output should look like:
+
+    <master/tokens> generated token: "f0c861.753c505740ecde4c"
+    <master/pki> created keys and certificates in "/etc/kubernetes/pki"
+    <util/kubeconfig> created "/etc/kubernetes/kubelet.conf"
+    <util/kubeconfig> created "/etc/kubernetes/admin.conf"
+    <master/apiclient> created API client configuration
+    <master/apiclient> created API client, waiting for the control plane to become ready
+    <master/apiclient> all control plane components are healthy after 61.346626 seconds
+    <master/apiclient> waiting for at least one node to register and become ready
+    <master/apiclient> first node is ready after 4.506807 seconds
+    <master/discovery> created essential addon: kube-discovery
```

Comment:
this is going away as soon as kube-discovery gets merged into the apiserver, so I don't think we need to think too hard about this line of output right now